test/unit/gateways/s5_test.rb

```diff
@@ -0,0 +1,413 @@
+require 'test_helper'
+
+class S5Test < Test::Unit::TestCase
+  def setup
+    @gateway = S5Gateway.new(
+      sender: 'sender',
+      channel: 'channel',
+      login: 'login',
+      password: 'password'
+    )
+
+    @credit_card = credit_card
+    @amount = 100
+
+    @options = {
+      order_id: '1',
+      billing_address: address,
+      description: 'Store Purchase'
+    }
+  end
+
+  def test_successful_purchase
+    @gateway.expects(:ssl_post).returns(successful_purchase_response)
+
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+
+    assert_equal '8a8294494d0a8ecd014d25a71d1502c7', response.authorization
+    assert response.test?
+  end
+
+  def test_failed_purchase
+    @gateway.expects(:ssl_post).returns(failed_purchase_response)
+
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_failure response
+  end
+
+  def test_successful_authorize
+    @gateway.expects(:ssl_post).returns(successful_authorize_response)
+
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+  end
+
+  def test_failed_authorize
+    @gateway.expects(:ssl_post).returns(failed_authorize_response)
+
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_failure auth
+  end
+
+  def test_successful_capture
+    @gateway.expects(:ssl_post).returns(successful_capture_response)
+
+    assert response = @gateway.capture(@amount, '8a8294494d0a8ecd014d25a71d1502c7')
+    assert_success response
+    assert_match %r{Request successfully processed}, response.message
+  end
+
+  def test_failed_capture
+    @gateway.expects(:ssl_post).returns(failed_capture_response)
+
+    response = @gateway.capture(nil, '')
+    assert_failure response
+  end
+
+  def test_successful_refund
+    @gateway.expects(:ssl_post).returns(successful_refund_response)
+
+    assert refund = @gateway.refund(@amount, '8a8294494d0a8ecd014d25a71d1502c7')
+    assert_success refund
+  end
+
+  def test_failed_refund
+    @gateway.expects(:ssl_post).returns(failed_refund_response)
+
+    response = @gateway.refund(nil, '')
+    assert_failure response
+  end
+
+  def test_successful_void
+    @gateway.expects(:ssl_post).returns(successful_void_response)
+
+    assert void = @gateway.void('8a8294494d0a8ecd014d25a71d1502c7')
+    assert_success void
+  end
+
+  def test_failed_void
+    @gateway.expects(:ssl_post).returns(failed_void_response)
+
+    response = @gateway.void('')
+    assert_failure response
+  end
+
+  def test_successful_verify
+    @gateway.expects(:ssl_post).twice.returns(successful_authorize_response)
+
+    response = @gateway.verify(@credit_card, @options)
+    assert_success response
+  end
+
+  def test_successful_verify_with_failed_void
+    @gateway.expects(:ssl_post).returns(failed_void_response)
+
+    response = @gateway.verify(@credit_card, @options)
+    assert_failure response
```

Comment:
Yup.  Since if the auth works and the void fails, the verify should be successful.